lib/managers/webhooks.js

```diff
@@ -193,6 +193,120 @@ Webhooks.prototype.delete = function(webhookID, callback) {
 	this.client.del(apiPath, null, this.client.defaultResponseHandler(callback));
 };
 
+const crypto = require('crypto');
+
+/**
+ * Compute the message signature (see https://docs.box.com/reference#signatures)
+ *
+ * @param {string} body - The request body of the webhook message
+ * @param {Object} headers - The request headers of the webhook message
+ * @param {string} signatureKey - The signature to verify the message with
+ * @returns {string} - The message signature
+ * @private
+ */
+function computeSignature(body, headers, signatureKey) {
+	if (!signatureKey) {
+		return undefined;
+	}
+
+	if (headers['box-signature-version'] !== '1') {
+		// console.log('Unrecognized signature version: %s', headers['box-signature-version']);
+		return undefined;
+	}
+
+	if (headers['box-signature-algorithm'] !== 'HmacSHA256') {
+		// console.log('Unrecognized signature algorithm: %s', headers['box-signature-algorithm']);
+		return undefined;
+	}
+
+	let hmac = crypto.createHmac('sha256', signatureKey);
+	hmac.update(body);
+	hmac.update(headers['box-delivery-timestamp']);
+
+	const signature = hmac.digest('base64');
+	// console.log('Signature: %s', signature);
+
+	return signature;
+}
+
+/**
+ * Validate the message signature (see https://docs.box.com/reference#signatures)
+ *
+ * @param {string} body - The request body of the webhook message
+ * @param {Object} headers - The request headers of the webhook message
+ * @param {string} primarySignatureKey - The primary signature to verify the message with
+ * @param {?string} secondarySignatureKey - The secondary signature to verify the message with
```

Comment:
This param isn't just nullable, it's totally optional!  You can denote that like `@param {string} [secondarySignatureKey]`